src/pytest_mock_resources/fixture/database/relational/generic.py

```diff
@@ -154,6 +155,59 @@ def manage(self, session=None):
         finally:
             self.engine.dispose()
 
+    def manage_sync(self, session=None):
+        try:
+            self._run_actions()
+
+            if session:
+                if isinstance(session, sessionmaker):
+                    session_factory = session
+                else:
+                    session_factory = sessionmaker(bind=self.engine)
+
+                Session = scoped_session(session_factory)
+                session = Session(bind=self.engine)
+                yield session
+                session.close()
+            else:
+                yield self.engine
+        finally:
+            self.engine.dispose()
+
+    def manage_async(self, session=None):
+        try:
+            self._run_actions()
+
+            engine = self._get_async_engine()
+
+            if session:
+                if isinstance(session, sessionmaker):
```

Comment:
i think it's worth copying the above if/session construction pattern above so that session is yielded at only 1 code point, so that you can `await session.close()` after that single point

Reply:
Similar to my comment below for engines. We haven't initialized a session yet so we don't need to close it.